File: backend/tests/fixtures/java_nested/src/main/java/com/example/nested/Nested.java
package com.example.nested;

public class Outer {

    public class Inner {
        public int run() {
            return step();
        }

        public int step() {
            return 1;
        }
    }

    public static class StaticNested {
        public String name() {
            return "static";
        }
    }

    public Inner make() {
        return new Inner();
    }
}

File: backend/tests/fixtures/python_nested/nested.py
"""Fixture exercising nested functions, nested classes, and dynamic calls (T-04)."""


def outer(x):
    def inner(y):
        return y * 2

    return inner(x) + helper()


def helper():
    return 1


class Wrapper:
    class Inner:
        def run(self):
            return self._step()

        def _step(self):
            label = getattr(self, "label", "none")
            return len(label)

    def make(self):
        return Wrapper.Inner()


def dynamic_caller(obj, method):
    return getattr(obj, method)()
